verifier/src/report_body.rs

```diff
@@ -2,72 +2,95 @@
 
 //! Verifiers which operate on the [`ReportBody`]
 
-use crate::{And, AndError, VerificationError, Verifier};
+use crate::{AlwaysTrue, AndError, VerificationError, Verifier};
 use mc_sgx_core_types::{Attributes, ConfigId, ReportBody};
 use subtle::CtOption;
 
 /// Verify the report body is as expected.
+#[derive(Clone, Debug, Eq, Hash, PartialEq, Default)]
+pub struct ReportBodyVerifier {
+    attributes_verifier: ReportBodyVerifierKind,
+    config_id_verifier: ReportBodyVerifierKind,
+}
+
 #[derive(Clone, Debug, Eq, Hash, PartialEq)]
-pub struct ReportBodyVerifier<'a> {
-    expected_report_body: ReportBody,
-    report_body: &'a ReportBody,
+pub enum ReportBodyVerifierKind {
+    AttributesVerifier(AttributesVerifier),
+    ConfigIdVerifier(ConfigIdVerifier),
+    AlwaysTrue(AlwaysTrue),
```

Comment:
As I was working on when/where the _evidence_ should be passed I realized that a client may only want to verify portions of `ReportBody`. Since the verification tree is a concrete type via generics it has to be known at compile time and thus can't be built up dynamically. In order to accomodate this I brought in the kind enum from the main mobilecoin repo.

I'm up for a more rusty alternative, keeping in mind that one of the desires was to have this be no_alloc.

Reply:
Doing things like this with an enum is plenty Rust-idiomatic. It might become tricky to work with if we then need to have the type system know something about this at compile-time later, but that seems unlikely given the purpose of this enum is so the actual concrete type can vary at run-time.

Reply:
This has now been removed, in favor of the the `Accessor` traite